python/beeai_framework/tools/mcp_tools.py

```diff
@@ -52,16 +43,17 @@ def is_empty(self) -> bool:
         return not self.result
 
 
-class MCPTool(Tool[MCPToolOutput]):
+class MCPTool(Tool[BaseModel]):
     """Tool implementation for Model Context Protocol."""
 
-    def __init__(self, client: ClientSession, tool: MCPToolInfo, **options: int) -> None:
+    def __init__(self, server_params: StdioServerParameters, tool: MCPToolInfo, **options: int) -> None:
         """Initialize MCPTool with client and tool configuration."""
         super().__init__(options)
-        self.client = client
         self._tool = tool
         self._name = tool.name
         self._description = tool.description or "No available description, use the tool based on its name and schema."
+        self.input_schema = json_to_model(self._name, tool.inputSchema)
+        self._server_params = server_params
 
     @property
     def name(self) -> str:
```

Comment:
Remove `self._name = tool.name` and do this instead.

@property
def name() -> str:
  return self._tool.name

@property
def description() -> str:
  return self._tool.description

@property
def input_schema() -> str:
  return json_to_model(self.name, self._tool.inputSchema)